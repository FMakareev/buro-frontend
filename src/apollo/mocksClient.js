import faker from 'faker';
import setupClient from './helpers/apolloClientMock';
import schema from './schema.graphqls';
import { userDocumentList } from './graphql/query/userDocumentList';
import { userList } from './graphql/query/userList';
import { userItem } from './graphql/query/userItem';
import { notificationList } from './graphql/query/notificationList';
import { ROLE_BANK, ROLE_BUREAU, ROLE_CLIENT } from '../shared/roles';

const defaultMocks = {
  Query: () => ({
    userList,
    userItem,
    userEmailItem: (query, { email }) => {
      console.log(query, email);
      switch (email) {
        case 'client@test.com': {
          return {
            ...userItem(),
            email: 'client@test.com',
            role: ROLE_CLIENT,
          };
        }
        case 'bank@test.com': {
          return {
            ...userItem(),
            email: 'bank@test.com',
            role: ROLE_BANK,
          };
        }
        case 'bureau@test.com': {
          return {
            ...userItem(),
            email: 'bureau@test.com',
            role: ROLE_BUREAU,
          };
        }
        default: {
          throw Error(
            JSON.stringify({
              userEmailItem: null,
              errors: [
                {
                  message: 'GraphQL error: user not found',
                },
              ],
            }),
          );
        }
      }
    },
    userDocumentList,
    notificationList,
  }),
  Mutation: () => ({
    /**
     * @params {func} mutation - этот же запрос
     * @params {object} props - аргументы которые были переданы
     * */
    createUser: (mutation, props) => props,
    updateUser: (mutation, props) => props,
    userResetPassword: (mutation, props) => props,
    createNotification: (mutation, props) =>
      // для имитации запроса к серверу с рандомной задержкой и результатом.
      new Promise((resolve, reject) => {
        setTimeout(() => {
          faker.random.number(1)
            ? resolve(JSON.stringify({ data: { createNotification: props } }))
            : reject(
                JSON.stringify({
                  errors: [
                    {
                      message: 'error!',
                    },
                  ],
                }),
              );
        }, faker.random.number(2000));
      }),
    updateNotification: (mutation, props) =>
      // для имитации запроса к серверу с рандомной задержкой и результатом.
      new Promise((resolve, reject) => {
        setTimeout(() => {
          faker.random.number(1)
            ? resolve(JSON.stringify({ data: { updateNotification: props } }))
            : reject(
                JSON.stringify({
                  errors: [
                    {
                      message: 'error!',
                    },
                  ],
                }),
              );
        }, faker.random.number(2000));
      }),
    userPasswordRecovery: (mutation, props) =>
      // throw Error(JSON.stringify({
      //   userPasswordRecovery: null,
      //   errors: [
      //     {
      //       message:'Error!'
      //     }
      //   ]
      // }))
      props,
  }),
};

export const mocksClient = setupClient(defaultMocks, schema)();

export default mocksClient;
