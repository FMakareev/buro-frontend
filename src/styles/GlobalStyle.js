import { createGlobalStyle } from 'styled-components';
import BG from '../assets/image/bg.jpg';

export const GlobalStyle = createGlobalStyle`
  *{
    box-sizing: border-box;
  }
  body, html, #app {
    padding: 0;
    margin: 0;
    width: 100%;
    min-height: 100vh;
    background-color: #fff;
    font-family: 'Montserrat Regular';
    font-size: 16px;
  }
  #app{
    background-color: #fff;
    background-image: url(${BG});
    background-repeat: no-repeat;
    background-position: center;
    background-size: cover;
  }
  a {
    text-decoration: none;
    color: inherit;
  }
  
  button:focus, input:focus, select:focus {
    outline: none !important;
    box-shadow: 0 0 0 5px #E7EDF8;
  }
  

  @font-face {
    font-family: 'Montserrat Regular';
    src: url('../assets/fonts/Montserrat-Regular.ttf') format('truetype');
    font-weight: normal;
    font-style: normal;
  }

  @font-face {
  font-family: 'Montserrat Medium';
  src: url('../assets/fonts/Montserrat-Medium.ttf') format('truetype');
  font-weight: normal;
  font-style: normal;
  }

  @font-face {
  font-family: 'Montserrat Bold';
  src: url('../assets/fonts/Montserrat-Bold.ttf') format('truetype');
  font-weight: bold;
  font-style: normal;
}

`;

export default GlobalStyle;
