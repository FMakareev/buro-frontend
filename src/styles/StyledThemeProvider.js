import React from 'react';
import PropTypes from 'prop-types';
import { Provider } from 'rebass';
import { ButtonVariant } from './variants/buttonVariant';
import { ButtonSize } from './variants/buttonSize';
import { TextVariant } from './variants/textVariant';

export const ColorPallet = {
  color0: '#FFFFFF',
  color1: '#093971',
  color2: '#EFF9FC',
  color3: '#E7EDF8',
  color4: '#61C3E2',
  color5: '#A2A2A2',
  color6: '#0F60BD',
  color7: '#072C57',
  color8: '#61C3E2',
  color9: '#EB5757',
  color10: '#E7EDF8',
  color11: '',
};

const Space = [
  0,
  2,
  4,
  8,
  12,
  14,
  16,
  18,
  20,
  24,
  28,
  32,
  36,
  40,
  44,
  48,
  52,
  56,
  60,
  64,
  68,
  72,
  76,
  80,
];

const Weight = [300, 500, 700];

const Breakpoints = ['576px', '768px', '992px', '1200px'];

const boxShadow = ['4px 8px 16px rgba(28, 65, 84, 0.08)'];

export const StyledThemeProvider = ({ children }) => (
  <Provider
    theme={{
      space: Space,
      fontSizes: Space,
      fontWeight: Weight,
      lineHeight: Space,
      boxShadow,
      borderRadius: Space,
      borderColor: ColorPallet,
      breakpoints: Breakpoints,
      colors: ColorPallet,
      variant: {
        buttons: ButtonVariant,
        buttonSize: ButtonSize,
        text: TextVariant,
      },
      fontFamily: {
        regular: 'Montserrat Regular',
        medium: 'Montserrat Medium',
        bold: 'Montserrat Bold',
      },
    }}>
    {children}
  </Provider>
);

StyledThemeProvider.propTypes = {
  children: PropTypes.element.isRequired,
};

StyledThemeProvider.defaultProps = {
  children: 'children is empty',
};

export default StyledThemeProvider;
