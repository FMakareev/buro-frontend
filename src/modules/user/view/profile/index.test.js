import React from 'react';
import renderer from 'react-test-renderer';
import {ApolloProvider} from 'react-apollo';
import wait from 'waait';
import mocksClient from "../../../../apollo/mocksClient";

import {MemoryRouter, Redirect} from 'react-router-dom';
import {ProfilePage} from './index';
import {StyledThemeProvider} from "../../../../styles/StyledThemeProvider";
import {Provider as ProviderRedux} from 'react-redux';
import configureStore from 'redux-mock-store'


const mockStore = configureStore();


test('ProfilePage: пользователь не авторизован', () => {

  const store = mockStore({});

  const output = renderer.create(<StyledThemeProvider>
    <ProviderRedux store={store}>
      <MemoryRouter>
        <ApolloProvider client={mocksClient}>
          <ProfilePage/>
        </ApolloProvider>
      </MemoryRouter>
    </ProviderRedux>
  </StyledThemeProvider>);
  expect(output.root.findByType(Redirect).props.to).toBe('/logout');
});


test('ProfilePage: загрузка пользовательских данных', () => {
  const store = mockStore({
    user: {
      error: null,
      initLoading: false,
      updateLoading: false,
      role: [],
      email: 'client@test.com',
    }
  });

  const output = renderer.create(<StyledThemeProvider>
    <ProviderRedux store={store}>
      <MemoryRouter>
        <ApolloProvider client={mocksClient}>
          <ProfilePage/>
        </ApolloProvider>
      </MemoryRouter>
    </ProviderRedux>
  </StyledThemeProvider>);
  expect(output).toMatchSnapshot();
});

test('ProfilePage: ошибка во время загрузки пользовательских данных', async () => {
  const store = mockStore({
    user: {
      error: null,
      initLoading: false,
      updateLoading: false,
      role: [],
      email: 'null',
    }
  });

  const output = renderer.create(<StyledThemeProvider>
    <ProviderRedux store={store}>
      <MemoryRouter>
        <ApolloProvider client={mocksClient}>
          <ProfilePage/>
        </ApolloProvider>
      </MemoryRouter>
    </ProviderRedux>
  </StyledThemeProvider>);

  await wait(2); // небольшая задержка

  expect(output).toMatchSnapshot();
});

test('ProfilePage: загруженный профиль', async () => {
  const store = mockStore({
    user: {
      error: null,
      initLoading: false,
      updateLoading: false,
      role: [],
      email: 'client@test.com',
    }
  });

  const output = renderer.create(<StyledThemeProvider>
    <ProviderRedux store={store}>
      <MemoryRouter>
        <ApolloProvider client={mocksClient}>
          <ProfilePage/>
        </ApolloProvider>
      </MemoryRouter>
    </ProviderRedux>
  </StyledThemeProvider>);
  await wait(2); // небольшая задержка

  expect(output).toMatchSnapshot();
});
