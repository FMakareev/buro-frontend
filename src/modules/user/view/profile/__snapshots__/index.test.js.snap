// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfilePage: загруженный профиль 1`] = `
<div
  className="sc-bxivhb dUiVDW"
>
  <div
    className="sc-1jxwge6-0 sc-14k9gvj-0 mQmih"
  >
    <form
      onSubmit={[Function]}
    >
      <h2
        className="sc-1jxwge6-0 sc-18ch56c-0 jEovzQ"
        color="color1"
        fontFamily="bold"
        fontSize={9}
      >
        Main data:
      </h2>
      <div
        className="sc-1jxwge6-0 wterhv-0 bBNfoQ"
      >
        <div
          className="sc-1jxwge6-0 iDrvMd"
          order={
            Array [
              5,
              0,
            ]
          }
          width={
            Array [
              "100%",
              "100%",
              "50%",
            ]
          }
        >
          <div
            className="uh5600-0 bTQmIh"
          >
            <div
              className="sc-1jxwge6-0 sc-18ch56c-0 sc-1hfdosk-0 cNcynQ"
              color="color1"
              fontFamily="regular"
              fontSize={5}
            >
              Telephone:
            </div>
            <input
              aria-required={false}
              className="uh5600-1 hRLQUv wsrc07-0 kZIRDH"
              label="Telephone:"
              name="phone"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="sc-1jxwge6-0 wterhv-0 eLvilD"
      >
        <div
          className="sc-1jxwge6-0 gl5g86-0 cUSPHQ"
        >
          <button
            className="mxzgdr-0 hhHYBZ"
            disabled={true}
            size="medium"
            type="submit"
          >
            Save
          </button>
        </div>
      </div>
    </form>
    <form
      onSubmit={[Function]}
    >
      <h2
        className="sc-1jxwge6-0 sc-18ch56c-0 gQJYko"
        color="color1"
        fontFamily="bold"
        fontSize={9}
      >
        Change password:
      </h2>
      <div
        className="sc-1jxwge6-0 wterhv-0 jLThhd"
      >
        <div
          className="sc-1jxwge6-0 ekfykU"
          width={
            Array [
              "100%",
              "100%",
              "50%",
            ]
          }
        >
          <div
            className="uh5600-0 bTQmIh"
          >
            <div
              className="sc-1jxwge6-0 sc-18ch56c-0 sc-1hfdosk-0 cNcynQ"
              color="color1"
              fontFamily="regular"
              fontSize={5}
            >
              Old Password:
            </div>
            <input
              aria-required={false}
              className="uh5600-1 hRLQUv wsrc07-0 kZIRDH"
              label="Old Password:"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          className="sc-1jxwge6-0 ekfykU"
          width={
            Array [
              "100%",
              "100%",
              "50%",
            ]
          }
        >
          <div
            className="uh5600-0 bTQmIh"
          >
            <div
              className="sc-1jxwge6-0 sc-18ch56c-0 sc-1hfdosk-0 cNcynQ"
              color="color1"
              fontFamily="regular"
              fontSize={5}
            >
              New password:
            </div>
            <input
              aria-required={false}
              className="uh5600-1 hRLQUv wsrc07-0 kZIRDH"
              label="New password:"
              name="newPassword"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          className="sc-1jxwge6-0 jZaAlL"
          width={
            Array [
              "100%",
              "100%",
              "50%",
            ]
          }
        >
          <div
            className="uh5600-0 bTQmIh"
          >
            <div
              className="sc-1jxwge6-0 sc-18ch56c-0 sc-1hfdosk-0 cNcynQ"
              color="color1"
              fontFamily="regular"
              fontSize={5}
            >
              Confirm new password:
            </div>
            <input
              aria-required={false}
              className="uh5600-1 hRLQUv wsrc07-0 kZIRDH"
              label="Confirm new password:"
              name="confirmPassword"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="sc-1jxwge6-0 wterhv-0 budtLR"
      >
        <div
          className="sc-1jxwge6-0 sc-6ivhj6-0 bbKauW"
          width={
            Array [
              "100%",
              "100%",
              "50%",
            ]
          }
        >
          <button
            className="mxzgdr-0 dypMfz"
            disabled={true}
            display="inline-block"
            size="medium"
            type="submit"
          >
            Confirm
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`ProfilePage: загрузка пользовательских данных 1`] = `
<div
  className="sc-bxivhb dUiVDW"
>
  <div
    className="sc-1jxwge6-0 sc-14k9gvj-0 mQmih"
  >
    Loading..,
    <form
      onSubmit={[Function]}
    >
      <h2
        className="sc-1jxwge6-0 sc-18ch56c-0 gQJYko"
        color="color1"
        fontFamily="bold"
        fontSize={9}
      >
        Change password:
      </h2>
      <div
        className="sc-1jxwge6-0 wterhv-0 jLThhd"
      >
        <div
          className="sc-1jxwge6-0 ekfykU"
          width={
            Array [
              "100%",
              "100%",
              "50%",
            ]
          }
        >
          <div
            className="uh5600-0 bTQmIh"
          >
            <div
              className="sc-1jxwge6-0 sc-18ch56c-0 sc-1hfdosk-0 cNcynQ"
              color="color1"
              fontFamily="regular"
              fontSize={5}
            >
              Old Password:
            </div>
            <input
              aria-required={false}
              className="uh5600-1 hRLQUv wsrc07-0 kZIRDH"
              label="Old Password:"
              name="password"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          className="sc-1jxwge6-0 ekfykU"
          width={
            Array [
              "100%",
              "100%",
              "50%",
            ]
          }
        >
          <div
            className="uh5600-0 bTQmIh"
          >
            <div
              className="sc-1jxwge6-0 sc-18ch56c-0 sc-1hfdosk-0 cNcynQ"
              color="color1"
              fontFamily="regular"
              fontSize={5}
            >
              New password:
            </div>
            <input
              aria-required={false}
              className="uh5600-1 hRLQUv wsrc07-0 kZIRDH"
              label="New password:"
              name="newPassword"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              type="password"
              value=""
            />
          </div>
        </div>
        <div
          className="sc-1jxwge6-0 jZaAlL"
          width={
            Array [
              "100%",
              "100%",
              "50%",
            ]
          }
        >
          <div
            className="uh5600-0 bTQmIh"
          >
            <div
              className="sc-1jxwge6-0 sc-18ch56c-0 sc-1hfdosk-0 cNcynQ"
              color="color1"
              fontFamily="regular"
              fontSize={5}
            >
              Confirm new password:
            </div>
            <input
              aria-required={false}
              className="uh5600-1 hRLQUv wsrc07-0 kZIRDH"
              label="Confirm new password:"
              name="confirmPassword"
              onBlur={[Function]}
              onChange={[Function]}
              onDragStart={[Function]}
              onDrop={[Function]}
              onFocus={[Function]}
              type="password"
              value=""
            />
          </div>
        </div>
      </div>
      <div
        className="sc-1jxwge6-0 wterhv-0 budtLR"
      >
        <div
          className="sc-1jxwge6-0 sc-6ivhj6-0 bbKauW"
          width={
            Array [
              "100%",
              "100%",
              "50%",
            ]
          }
        >
          <button
            className="mxzgdr-0 dypMfz"
            disabled={true}
            display="inline-block"
            size="medium"
            type="submit"
          >
            Confirm
          </button>
        </div>
      </div>
    </form>
  </div>
</div>
`;

exports[`ProfilePage: ошибка во время загрузки пользовательских данных 1`] = `
<div
  className="sc-bxivhb dUiVDW"
>
  <div
    className="sc-1jxwge6-0 dgupjs-0 r6fvyg-0 eIdhcX"
  >
    <div
      className="r6fvyg-1 frQDoI"
      color="color6"
      fill="currentColor"
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <svg
        fill="currentColor"
        height={64}
        style={
          Object {
            "display": "inline-block",
            "verticalAlign": "middle",
          }
        }
        viewBox="0 0 24 24"
        width={64}
      >
        <path
          d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
        />
      </svg>
    </div>
    <div
      className="sc-1jxwge6-0 sc-18ch56c-0 bghFjN"
      color="color1"
      fontSize={7}
    >
      error_unexpected_error_title
    </div>
    <div
      className="sc-1jxwge6-0 sc-18ch56c-0 dQGQvl"
      color="#000"
      fontSize={5}
    >
      unexpected error
    </div>
  </div>
</div>
`;
